alacritty_terminal/src/config/mod.rs

```diff
@@ -242,20 +243,47 @@ impl Default for EscapeChars {
 }
 
 #[serde(default)]
-#[derive(Deserialize, Copy, Clone, Debug, Default, PartialEq, Eq)]
+#[derive(Deserialize, Copy, Clone, Debug, PartialEq)]
 pub struct Cursor {
     #[serde(deserialize_with = "failure_default")]
     pub style: CursorStyle,
     #[serde(deserialize_with = "option_explicit_none")]
     pub vi_mode_style: Option<CursorStyle>,
+    #[serde(deserialize_with = "default_thickness")]
+    thickness: Percentage,
     #[serde(deserialize_with = "failure_default")]
     unfocused_hollow: DefaultTrueBool,
 }
 
 impl Cursor {
+    #[inline]
     pub fn unfocused_hollow(self) -> bool {
         self.unfocused_hollow.0
     }
+
+    #[inline]
+    pub fn thickness(self) -> f64 {
+        self.thickness.0 as f64
+    }
```

Comment:
Why do we cast this to `f64` here, rather than making `Percentage` an `f64`?

Precision isn't really very important with these options, but it seems extremely cheap in this context so I feel like bigger might sometimes be better?

Reply:
Alpha is f32 and we can't make it f64, so I've chosen to use this cast instead.